echo_server.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import unicode_literals
+import socket
+import os
+from mimetypes import guess_type
+
+ROOT_DIR = "root"
+
+
+def resolve_uri(uri):
+    path = "{}{}".format(ROOT_DIR, uri)
```

Comment:
You want to be careful about assembling filesystem paths like this.  Using `os.path.join` is more protected against shenanigans, though it's still not perfect.